include/ps/internal/message.h

```diff
@@ -64,15 +66,30 @@ struct Node {
   /** \brief the empty value */
   static const int kEmpty;
   /** \brief default constructor */
-  Node() : id(kEmpty), port(kEmpty) {}
+  Node() : id(kEmpty), port(kEmpty), is_recovery(false) {}
+  /** \brief copy constructor */
+  Node(const Node &other) : role(other.role), id(other.id),
+    hostname(other.hostname), port(other.port),
+    is_recovery(static_cast<bool>(other.is_recovery)) {}
+  /** \brief = operator overload */
+  Node& operator=(const Node& other) {
```

Comment:
is it necessary to write a copy constructor? c++ will automatically generate one.